packages/controllers/src/plugins/PluginController.ts

```diff
@@ -1,27 +1,27 @@
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
 import { ObservableStore } from '@metamask/obs-store';
-import EventEmitter from '@metamask/safe-event-emitter';
 import { ethErrors, serializeError } from 'eth-rpc-errors';
 import { IOcapLdCapability } from 'rpc-cap/dist/src/@types/ocap-ld';
-import { IRequestedPermissions } from 'rpc-cap/dist/src/@types';
 import { nanoid } from 'nanoid';
-import {
-  WorkerController,
-  SetupWorkerConnection,
-} from '../workers/WorkerController';
+import { BaseController } from '@metamask/controllers/dist/BaseControllerV2';
+
+import { RestrictedControllerMessenger } from '@metamask/controllers/dist/ControllerMessenger';
+import { Json, JsonRpcRequest } from 'json-rpc-engine';
+import { PluginData } from '@mm-snap/types';
+import { PluginWorkerMetadata } from '../workers/WorkerController';
 import { INLINE_PLUGINS } from './inlinePlugins';
 
 export const PLUGIN_PREFIX = 'wallet_plugin_';
 export const PLUGIN_PREFIX_REGEX = new RegExp(`^${PLUGIN_PREFIX}`, 'u');
-
 const SERIALIZABLE_PLUGIN_PROPERTIES = new Set([
   'initialPermissions',
   'name',
   'permissionName',
 ]);
 
-interface InititalPermissions {
-  [permission: string]: Record<string, unknown>;
-}
+type InititalPermissions = {
```

Comment:
Nit: should probably rename this type since it's being used for more than just initial permissions? I think this is something you mentioned on the call.